### PR TITLE
Add codespell to fix formatting

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
Add spell checker to fix formatting

### Testing Done
Ran manually, CI should verify if everything works as expected

### Resolved Issues
#1501 

### Length Justification
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by EvanMorcom — DISMISSED** [review]
LGTM 
🅱️ eauty

File: formatting_scripts/fix_formatting.sh [hunk 1]
```diff
@@ -64,7 +64,14 @@ function run_black_formatting () {
     fi
 }
 
+function run_code_spell(){
+    echo "Fixing spellings..."
+    cd $CURR_DIR/../src/software && codespell -w
```
Comment (MathewMacDougall) [inline formatting_scripts/fix_formatting.sh:69]: maybe do the `shared` directory too?

**Review by MathewMacDougall — APPROVED** [review]

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Whoop wrong button

File: formatting_scripts/fix_formatting.sh (hunk 1 above)
Comment (jonathanlew) [reply formatting_scripts/fix_formatting.sh:69]: Also why not firmware/app?

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Whoops didn't request changes

File: formatting_scripts/fix_formatting.sh (hunk 1 above)
Comment (akhilveeraghanta) [reply formatting_scripts/fix_formatting.sh:69]: done

**Review by jonathanlew — APPROVED** [review]
:snowflake: spell

**Review by MathewMacDougall — APPROVED** [review]

**Review by kelvinkoon — APPROVED** [review]

**Pull request merged**